metadata-ingestion/src/datahub/ingestion/source/state/entity_removal_state.py

```diff
@@ -29,6 +42,8 @@ def get_urns_not_in(
         # To maintain backwards compatibility, we provide this filtering mechanism.
         if type == "*":
             yield from diff
+        elif type == "topic":
+            yield from (urn for urn in diff if guess_entity_type(urn) == "dataset")
```

Comment:
Wow, these exceptions are scary

Reply:
they're only for compatibility with our existing tests - we don't need them in our actual code